From afc39d6f627f1e870883e8a6851f7a20f9700801 Mon Sep 17 00:00:00 2001
From: sangxu <sangxu@huawei.com>
Date: Wed, 30 May 2018 15:54:53 +0800
Subject: [PATCH]fix default file zero after power outage, fsync file immediately and use correct parameters info when removing old file. 

---
 usr/idbm.c | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/usr/idbm.c b/usr/idbm.c
index 5437f45..c01c4fe 100644
--- a/usr/idbm.c
+++ b/usr/idbm.c
@@ -2074,6 +2074,23 @@ mkdir_portal:
 	}
 
 	idbm_print(IDBM_PRINT_TYPE_NODE, rec, 1, f);
+
+	rc = fflush(f);
+	if (rc){
+		log_error("Could not fflush %s: %s\n", portal, strerror(errno));
+		rc = ISCSI_ERR_IDBM;
+		fclose(f);
+		goto free_portal;
+
+	}
+
+	if (fsync(fileno(f)) < 0){
+		log_error("Could not fsync %s: %s\n", portal, strerror(errno));
+		rc = ISCSI_ERR_IDBM;
+		fclose(f);
+		goto free_portal;
+	 }
+
 	fclose(f);
 
 	/* add for default_bak -> default */
@@ -2392,7 +2409,7 @@ int idbm_add_node(node_rec_t *newrec, discovery_rec_t *drec, int overwrite)
 		if (!overwrite)
 			return 0;
 
-		rc = idbm_delete_node(&rec);
+		rc = idbm_delete_node(newrec);
 		if (rc)
 			return rc;
 		log_debug(7, "overwriting existing record");
-- 
1.8.3.1

