From 65c06d977a012541fd6144d4452f5191081fdd44 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Wed, 30 Oct 2019 12:11:42 -0700
Subject: [PATCH 26/43] Uninitialized scalar variable

---
 usr/host.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/host.c b/usr/host.c
index 62bf111..356650c 100644
--- a/usr/host.c
+++ b/usr/host.c
@@ -217,7 +217,7 @@ static int print_host_iface(void *data, struct iface_rec *iface)
 
 static void print_host_ifaces(struct host_info *hinfo, char *prefix)
 {
-	int nr_found;
+	int nr_found = 0;
 
 	iscsi_sysfs_for_each_iface_on_host(prefix, hinfo->host_no, &nr_found,
 					   print_host_iface);
-- 
2.21.0

